test/fuzz/random.h

```diff
@@ -0,0 +1,101 @@
+#include <memory>
+#include <random>
+#include <utility>
+#include <vector>
+
+#include "envoy/common/random_generator.h"
+
+#include "common/common/assert.h"
+#include "common/common/logger.h"
+
+namespace Envoy {
+namespace Random {
+
+class PsuedoRandomGenerator64 : public RandomGenerator {
+public:
+  PsuedoRandomGenerator64() = default;
+  ~PsuedoRandomGenerator64() override = default;
+
+  void initializeSeed(uint64_t seed) { prng_ = std::make_unique<std::mt19937_64>(seed); }
+
+  // RandomGenerator
+  uint64_t random() override {
+    // Makes sure initializeSeed() was already called
+    ASSERT(prng_ != nullptr);
+    uint64_t to_return = (*prng_)();
+    ENVOY_LOG_MISC(trace, "random() returned: {}", to_return);
+    return to_return;
+  }
+  std::string uuid() override { return ""; }
+  std::unique_ptr<std::mt19937_64> prng_;
+};
+
+} // namespace Random
+
+namespace Fuzz {
+class ProperSubsetSelector {
```

Comment:
This now looks really good. I'd suggest a short gtest for this as well.

Reply:
Added.